src/vm/assemblyspec.cpp

```diff
@@ -1767,6 +1799,40 @@ BOOL AssemblySpecBindingCache::StoreException(AssemblySpec *pSpec, Exception* pE
     }
 }
 
+BOOL AssemblySpecBindingCache::RemoveAssembly(DomainAssembly* pAssembly)
```

Comment:
This appears to be logic for removing a DomainAssembly from the AssemblySpecBindingCache when it is unloaded, but I don't see the case where AssemblyLoadContext 1 has a reference to an assembly in AssemblyLoadContext 2, and we unload AssemblyLoadContext 1.

Reply:
I believe this should be ok. The `RemoveAssembly` is called from `LoaderAllocator::GCLoaderAllocators_RemoveAssemblies` and we only call it for assemblies that are loaded in the `AssemblyLoadContext` that's being destroyed. They are taken from the `m_pFirstDomainAssemblyFromSameALCToDelete` list.

Reply:
That's precisely my point. That isn't covering all the cases in which you should need to delete things.